pkg/database/metrics.go

```diff
@@ -0,0 +1,46 @@
+package database
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"time"
+
+	"github.com/Azure/ARO-RP/pkg/database/cosmosdb"
+)
+
+func (db Database) emitMetrics(ctx context.Context) {
+	t := time.NewTicker(time.Minute)
+	defer t.Stop()
+
+	for {
+		query := cosmosdb.Query{
+			Query: `SELECT * FROM OpenShiftClusters doc WHERE NOT (doc.openShiftCluster.properties.provisioningState IN ("Succeeded", "Failed")) AND (doc.leaseExpires ?? 0) < GetCurrentTimestamp() / 1000`,
```

Comment:
needs to go into pkg/database/openshiftclusters.go.  Replace the GetRaw fn with something like QueueLength().
use `SELECT VALUE COUNT(*) FROM ...` to get the count instead of trawling all the records

Reply:
There is some caveats with this... Will update